### PR TITLE
add isContractAddress to check for contract vs EOA addresses

### PR DESCRIPTION
**Pull request by SahilAujla**
Notion doc about it [here](https://www.notion.so/alchemotion/2023-06-isContractAddress-SDK-endpoint-c837160fc6d845f2a86f7f4614f81491?d=f824d416e7f040199916eed8bb40f54e)

isContractAddress-tests

### CODE REVIEW

File: src/api/core-namespace.ts [hunk 1]
```diff
@@ -73,6 +73,18 @@ export class CoreNamespace {
     return provider.getBalance(addressOrName, blockTag);
   }
 
+  /**
+   * Checks if the provided address is a smart contract.
+   *
+   * @param address The address to check type for.
+   * @public
+   */
+  async isContractAddress(address: string | Promise<string>): Promise<boolean> {
```
[the remark this answers is not in the document]
Comment (SahilAujla) [reply src/api/core-namespace.ts]: Done ✅

File: test/unit/core-namespace.test.ts [hunk 2]
```diff
@@ -13,4 +14,20 @@ describe('Core Namespace', () => {
       ).rejects.toThrow();
     });
   });
+
+  describe('isContractAddress()', () => {
```
[the remark this answers is not in the document]
Comment (SahilAujla) [reply test/unit/core-namespace.test.ts]: Done ✅

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,7 @@
 ### Major Changes
 
 - Fixed a bug with in `NftNamespace.getNftSales()` in which there was a typo in `decimal` field of the `NftSaleFeeData` object. The field is now correctly named `decimals`, and `decimal` is marked deprecated to avoid a breaking change.
+- Added a new method `CoreNamespace.isContractAddress()` to easily check if an address is a contract or EOA. The method accepts a string address and returns a boolean value indicating if the given address is a contract address or not.
```
Comment (thebrianchen) [inline CHANGELOG.md:14]: can you move this to the `Unreleased` section?

Comment (SahilAujla) [reply CHANGELOG.md]: Done!

**Pull request merged**